homeassistant/components/joaoapps_join.py

```diff
@@ -19,26 +19,19 @@
 CONF_DEVICE_ID = 'device_id'
 
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [{
         vol.Required(CONF_DEVICE_ID): cv.string,
         vol.Optional(CONF_NAME): cv.string,
         vol.Optional(CONF_API_KEY): cv.string
```

Comment:
There could be situations where you would need multiple api keys. For
example if you wanted to use groups for multiple accounts.

On Tue, Jul 19, 2016, 3:17 PM Johann Kellerman <notifications@github.com>
wrote:

> In homeassistant/components/joaoapps_join.py
> <https://github.com/home-assistant/home-assistant/pull/2567#discussion_r71404946>
> :
>
> > @@ -19,26 +19,19 @@
> >  CONF_DEVICE_ID = 'device_id'
> >
> >  CONFIG_SCHEMA = vol.Schema({
> > -    DOMAIN: vol.Schema({
> > +    DOMAIN: vol.All(cv.ensure_list, [{
> >          vol.Required(CONF_DEVICE_ID): cv.string,
> >          vol.Optional(CONF_NAME): cv.string,
> >          vol.Optional(CONF_API_KEY): cv.string
>
> Do you need multiple API keys?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2567/files/302c7e74bf4223a99dcb03e7924391b6176126df#r71404946>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABcAqj_IMkYilBS49hvRTLBLkZ3Kadgwks5qXSLjgaJpZM4JQBGV>
> .
>